src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.Create.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography
+{
+    partial class HashAlgorithm
+    {
+        public static HashAlgorithm Create() =>
+            throw new PlatformNotSupportedException(SR.Cryptography_DefaultAlgorithm_NotSupported);
```

Comment:
@bartonjs, remind me why we throw PNSE here rather than returning something as we do in Create(string)?

@baulig, on second thought, rather than doing any ifdef'ing in this file, since these APIs are already calling out to this CryptoConfigForwarder internal class, could we just leave intact, change the Create() overload to also call into CryptoConfigForwarder, and then add a similar internal type to Mono?  We could rename CryptoConfigForwarder if there's a better name for it.

Reply:
> remind me why we throw PNSE here rather than returning something as we do in Create(string)?

This method means "give me some sort of hash algorithm, I don't care what".  There's no user scenario for that, so we PNSE.

Every context where a hash is needed it's important to know what hash is being built.

Reply:
>  could we just leave intact, change the Create() overload to also call into CryptoConfigForwarder

Nope.

Reply:
> Nope.

I'm not saying it needs to forward... whatever method it calls could throw in core just as it's doing, and do whatever Mono wants to do in Mono's implementation.

Reply:
> Every context where a hash is needed it's important to know what hash is being built.

I see usage of it where context isn't particularly important, places where code just wants to hash something for non-security reasons, e.g. using it to generate a Guid tied to some input.  Why break such uses?  If it's important to know what hash is being built, the developer can specify.

Reply:
Well, mainly it's this:

* On netfx it returned SHA-1.
* SHA-1 is no longer a cryptographic hash, so it should not be returned as a cryptographic default
* Changing the default will break anyone who compares the value across runs, or an IPC boundary
* The change was too risky for netfx
* But we changed it here to say "we don't do defaults for cryptographic algorithm families; pick your algorithm".

Since we can't infer context, but this is in a Security namespace, we're secure by default... and that means "never ever call this method" => PNSE.

Reply:
> I'm not saying it needs to actually "forward"... whatever method it calls could throw in core just as it's doing, and do whatever Mono wants to do in Mono's implementation. Hence suggesting there may be a better name.

Ah.  Something like  `public static void Create() => CryptoConfigForwarder.CreateDefaultHashAlgorithm();`, and the corefx implementation of that is this PNSE?  Sure, that indirection is fine.

I read the comment as "forward to the other Create", a la the netfx implementation.

Reply:
> Ah. Something like public static void Create() => CryptoConfigForwarder.CreateDefaultHashAlgorithm();, and the corefx implementation of that is this PNSE? Sure, that indirection is fine.

Exactly.

Reply:
That's... probably cleaner overall.

Good thinking, @stephentoub. :wink: